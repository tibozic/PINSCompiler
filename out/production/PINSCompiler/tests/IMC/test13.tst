fun f(a: integer): integer = g(5)
	{ where fun g(b: integer): integer = h(a)
		{ where fun h(c: integer): integer = a }
	}
