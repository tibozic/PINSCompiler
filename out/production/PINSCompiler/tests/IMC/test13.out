FRAME [L[1]]: level=3,locals_size=0,arguments_size=0,parameters_size=8,size=8
MOVE:
  MEM:
    NAME: {FP}
  MEM:
    BINOP ADD:
      NAME: {FP}
      CONSTANT: 4
FRAME [L[0]]: level=2,locals_size=0,arguments_size=8,parameters_size=8,size=12
MOVE:
  MEM:
    NAME: {FP}
  ESEQ:
    MOVE:
      MEM:
        BINOP SUB:
          NAME: {SP}
          CONSTANT: 4
      NAME: {FP}
    CALL L[1]:
      MEM:
        NAME: {FP}
      MEM:
        BINOP ADD:
          NAME: {FP}
          CONSTANT: 4
FRAME [f]: level=1,locals_size=0,arguments_size=8,parameters_size=8,size=12
MOVE:
  MEM:
    NAME: {FP}
  ESEQ:
    MOVE:
      MEM:
        BINOP SUB:
          NAME: {SP}
          CONSTANT: 4
      NAME: {FP}
    CALL L[0]:
      CONSTANT: 0
      CONSTANT: 5
